ee/tabby-webserver/src/service/mod.rs

```diff
@@ -325,6 +341,8 @@ impl ServiceLocator for ArcServerContext {
 
 pub async fn create_service_locator(
     logger: Arc<dyn EventLogger>,
+    chat: Option<Arc<dyn ChatCompletionStream>>,
+    completion: Option<Arc<CodeGeneration>>,
```

Comment:
```suggestion
    completion: Option<Arc<CompletionStream>>,
```

`CodeGeneration` is a wrapper interface - use CompletionStream directly.